modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/DataLifecycleServiceIT.java

```diff
@@ -452,7 +451,16 @@ public void testErrorRecordingOnRollover() throws Exception {
             Iterable<DataLifecycleService> lifecycleServices = internalCluster().getInstances(DataLifecycleService.class);
 
             for (DataLifecycleService lifecycleService : lifecycleServices) {
-                assertThat(lifecycleService.getErrorStore().getError(previousWriteInddex), nullValue());
+                if (internalCluster().numDataNodes() > 1) {
+                    assertThat(lifecycleService.getErrorStore().getError(previousWriteInddex), nullValue());
+                } else {
+                    if (lifecycleService.getErrorStore().getError(previousWriteInddex) != null) {
+                        assertThat(
+                            lifecycleService.getErrorStore().getError(previousWriteInddex),
+                            containsString("Force merge request only had 1 successful shards out of a total of 2")
```

Comment:
So, this test is a called `testErrorRecordingOnRollover` and it is checking how rollover related errors are recorded and cleared up, right?

If my above hypothesis is correct, I have to admit that this last part confuses me a bit. If I understand correctly the issue here is that if the cluster has a single node then the replica shard is unassigned and the force merge fails right?

What if, we use `auto_expand_replicas: 0-1` or something like that to ensure there will be no other errors. My concern is that this logic isn't related to this test and it will only confuse future maintainers.

What do you think?

Reply:
Yeah I had originally thought about changing it to use a minimum cluster size of 2 to avoid this problem. But that seemed to be too much of an impact on the test. Using `auto_expand_replicas` seems like a good balance. I'll switch to that.